src/include/utime.h

```diff
@@ -400,17 +381,45 @@ class utime_t {
 
     const char *p = strptime(date.c_str(), "%Y-%m-%d", &tm);
     if (p) {
-      if (*p == ' ') {
+      if (*p == ' ' || *p == 'T') {
 	p++;
-	p = strptime(p, " %H:%M:%S", &tm);
-	if (!p)
+	// strptime doesn't understand fractional/decimal seconds, and
+	// it also only takes format chars or literals, so we have to
+	// get creative.
+	char fmt[32] = {0};
+	strncpy(fmt, p, sizeof(fmt) - 1);
+	fmt[0] = '%';
+	fmt[1] = 'H';
+	fmt[2] = ':';
+	fmt[3] = '%';
+	fmt[4] = 'M';
+	fmt[6] = '%';
+	fmt[7] = 'S';
+	const char *subsec = 0;
+	char *q = fmt + 8;
+	if (*q == '.') {
+	  ++q;
+	  subsec = p + 9;
+	  q = fmt + 9;
+	  while (*q && isdigit(*q)) {
+	    ++q;
+	  }
+	}
+	// look for tz...
+	if (*q == '-' || *q == '+' || *q == 'Z') {
+	  *q = '%';
+	  *(q+1) = 'z';
+	  *(q+2) = 0;
+	}
+	p = strptime(p, fmt, &tm);
```

Comment:
i'm seeing this call fail on centos, using a simple unit test from commit https://github.com/cbodley/ceph/commit/0b6eecfbbb23130e3560b2b113d028966c5a5f11:
```
414             p = strptime(p, fmt, &tm);
(gdb) p p
$17 = 0x5555558acaf3 "16:53:20.181200Z"
(gdb) p fmt
$18 = "%H:%M:%S.181200%z", '\000' <repeats 14 times>
(gdb) n
415             if (!p) {
(gdb) p p
$19 = 0x0
```